include/interface/drivers/IOExpander.h

```diff
@@ -0,0 +1,30 @@
+// Leka - LekaOS
+// Copyright 2021 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef _LEKA_OS_INTERFACE_DRIVER_IO_EXPANDER_H_
+#define _LEKA_OS_INTERFACE_DRIVER_IO_EXPANDER_H_
+
+#include <cstdint>
+
+#include "PinNames.h"
+
+namespace leka::interface {
+
+template <typename PinType>
+class IOExpander
+{
+  public:
+	virtual ~IOExpander() = default;
+
+	virtual void setPinAsInput(PinType pin) = 0;
+
+	virtual auto readInputPin(PinType pin) -> int = 0;
```

Comment:
Que ce soit en Input ou Output, ça ne change pas son comportement

```suggestion
	virtual auto readPin(PinType pin) -> int = 0;
```